bpf/attach.go

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bpf
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type AttachedProgInfo struct {
+	Object string `json:"object"`
+	Hash   string `json:"hash"`
+}
+
+// Check if a hash file exists for an Attach Point and the content matches
+// the Attach Point's fields
+func VerifyProgHash(iface, hook, object string) (bool, string, error) {
```

Comment:
`sha256OfFile()` is computing the hash value. I was trying to not compute the hash twice, but since the code is not executed many times, maybe it is it Ok to compute two time. I fixed the return values, and made `SaveProgHash` to calculate the hash for itself. How does it sound?